index.js

```diff
@@ -1,76 +1,119 @@
+    type: "string",
+    describe: "CSV filename in the current folder"
+  })
+  .option("app-id", {
+    type: "string",
+    describe: "AppID can be obtained from the Application Links menu in Jira (Click on pencil icon and ID is in the URL)"
+  })
+  .option("jira-auth", {
+    type: "string",
+    describe: "Username and password for the Jira user with 'Link Issues' permission in the following format 'username:password'"
+  })
+  .option("dry-run", {
+    type: "boolean",
+    default: false,
+    describe: "Script will not submit HTTP requests, if this flag is enabled."
+  })
+  .demandOption(["jiraUrl", "confUrl", "csv", "appId"])
+  .argv;
 
 const ISSUE_KEY_REGEX = /(?<=key&quot;&gt;)[A-Z]+-[0-9]+(?=&lt;)/g;
 
-// TODO Add feature for dry run - don't make HTTP requests
+const jiraEndpoint = argv['jira-url'];
+const confluenceEndpoint = argv['conf-url'];
+const csvFilename = argv['csv'];
+const appId = argv['app-id'];
+const usernameAndPassword = argv['jira-auth'];
+const isDryRun = argv['dry-run'];
 
-const jiraEndpoint = argv.jiraUrl;
-const confluenceEndpoint = argv.confUrl;
-const csvFilename = argv.csv;
-const appId = argv.appId;
+const requests = [];
+let createdLinksCount = 0;
+let preexistingLinksCount = 0;
+let errCount = 0;
 
-// Send HTTP requests to Jira
 const processRow = row => {
 
-    // Q: What about versions? Do we need to update on all versions?
+  const pageId = row["Page ID"];
+  const title = row["Page Title"];
+  const metadata = row["text_val"];
 
-    const pageId = row["Page ID"];
-    const title = row["Page Title"];
-    const metadata = row["text_val"];
+  const issueKeys = metadata.match(ISSUE_KEY_REGEX);
+  log(INFO, `[PAGE-ID: ${pageId}] Keys: ${issueKeys}`);
 
-    const issueKeys = metadata.match(ISSUE_KEY_REGEX);
-    console.log("Keys: ", issueKeys);
+  const body = {
+    globalId: `appId=${appId}&pageId=${pageId}`,
+    application: {
+      type: "com.atlassian.confluence",
+      name: "Confluence"
+    },
+    relationship: "mentioned in",
+    object: {
+      url: `${confluenceEndpoint}/pages/viewpage.action?pageId=${pageId}`,
+      title
+    }
+  };
 
-    const body = {
-        globalId: `appId=${appId}&pageId=${pageId}`,
-        application: {
-            type: "com.atlassian.confluence",
-            name: "Confluence"
-        },
-        relationship: "mentioned in",
-        object: {
-            url: `${confluenceEndpoint}/pages/viewpage.action?pageId=${pageId}`,
-            title
-        }
-    };
+  issueKeys.forEach(issueKey => {
+    const requestUrl = `${jiraEndpoint}/rest/api/latest/issue/${issueKey}/remotelink`;
 
-    console.log(body);
+    if (isDryRun) {
+      log(INFO, `[DRY-RUN] Will update ${requestUrl}`);
+      return;
+    }
 
-    issueKeys.forEach(issueKey => {
-        const requestUrl = `${jiraEndpoint}/rest/api/latest/issue/${issueKey}/remotelink`;
-        console.log(requestUrl);
-    });
+    const authBuffer = Buffer.from(usernameAndPassword, 'utf-8');
```

Comment:
The `usernameAndPassword` shouldn't change so I think it can be moved out of the `processRow` function.

Reply:
Addressed.